proto/storage/process_listening_on_port.proto

```diff
@@ -63,4 +63,6 @@ message ProcessListeningOnPortStorage {
     string deployment_id              = 8 [(gogoproto.moretags) = 'search:"Deployment ID,store" sql:"fk(Deployment:id),no-fk-constraint,index=btree,type(uuid)"'];
 
     string pod_uid                    = 9 [(gogoproto.moretags) = 'search:"Pod UID,hidden" sql:"fk(Pod:id),no-fk-constraint,index=hash,type(uuid)"'];
+
+    string cluster_id                 = 10 [(gogoproto.moretags) = 'search:"Cluster ID,store" sql:"type(uuid)"'];
```

Comment:
I am not 100% sure about this.

Reply:
I was looking at this some earlier and decided I had to come back to it.  This with the `cluster_id` is correct.  This will store the cluster_id in the table and apply the SAC checks.  Where this gets confusing is the other object `ProcessListeningOnPort` at the top of the file and how that one and this one interact.

Reply:
Can remove the store field from the search

Reply:
`alert.proto` for example has

```
    // Scoping information

    string                    cluster_id        = 18 [(gogoproto.moretags) = 'search:"Cluster ID,store" sql:"type(uuid)"'];
    string                    cluster_name      = 19 [(gogoproto.moretags) = 'search:"Cluster,store"'];
    string                    namespace         = 20 [(gogoproto.moretags) = 'search:"Namespace,store"'];
    string                    namespace_id      = 21 [(gogoproto.moretags) = 'search:"Namespace ID" sql:"type(uuid)"'];
```

Do we need all four of these fields? Doesn't look like other protos use all of these fields for scoping. This PR is only adding cluster_id anyway.

Reply:
@connorgorman Do you mean it should be

```suggestion
    string cluster_id                 = 10 [(gogoproto.moretags) = 'search:"Cluster ID" sql:"type(uuid)"'];
```

Reply:
yes that is what he is talking about.

Reply:
Regarding scoping `cluster_id` is the only field required for cluster scope.  `namespace` is required for namespace scope.  You can tell by looking at the generated gorm model to see where the `_sac_filter` index was automatically added.  For example:

```
	ClusterID                string                              `gorm:"column:clusterid;type:uuid;index:alerts_sac_filter,type:btree"`
	ClusterName              string                              `gorm:"column:clustername;type:varchar"`
	Namespace                string                              `gorm:"column:namespace;type:varchar;index:alerts_sac_filter,type:btree"`
	NamespaceID              string                              `gorm:"column:namespaceid;type:uuid"`
```

That does raise a question though, should these be namespace scoped as well?  @connorgorman

Reply:
There is another PR to add namespace to the listening_endpoints table. It is still in draft form as there are some failing tests https://github.com/stackrox/stackrox/pull/9182

Reply:
> I was looking at this some earlier and decided I had to come back to it. This with the cluster_id is correct. This will store the cluster_id in the table and apply the SAC checks. Where this gets confusing is the other object ProcessListeningOnPort at the top of the file and how that one and this one interact.

@dashrews78 Can you explain more about how the two might interact?

Reply:
@JoukoVirtanen that was my original question.  If I understand corrrectly,  `ProcessListeningOnPortStorage` is what is stored in the DB and `ProcessListeningOnPort` is what is returned to the user.  Ideally `ProcessListeningOnPort` would have been created in `proto/api/v1/process_listening_on_port_service.proto`.  

Can you put a comment on the objects in the proto stating their uses?

Reply:
I added some comments. I thought that there was some way that they might interact with regards to SAC that I wasn't aware of, and that is what you were referring to.